tests/libc_newlib/Makefile

```diff
@@ -5,5 +5,33 @@ USEMODULE += embunit
 
 include $(RIOTBASE)/Makefile.include
 
+# Compile time tests
+.PHONY: compile-test test-newlib-nano
+all: compile-test
+
+ifneq (,$(filter newlib,$(USEMODULE)))
+  COMPILE_TESTS += test-newlib
+  ifneq (,$(filter newlib_nano,$(USEMODULE)))
+    CMP_OP = =
+  else
+    CMP_OP = !=
+  endif
+endif
+
+test-newlib: $(ELFFILE)
```

Comment:
A dumber approach without sed (checks the number of different addresses)
```make
# make N_PRINTFS = 2 if non-nano, 1 if nano

test-newlib: $(ELFFILE)
   PRINTFS=$$($(NM) -v $^ | grep '\( i\?printf\)'); \
   PRINTF_ADDRS=$$(echo "$${PRINTFS}" | cut -d' ' -f 1 | uniq | wc -l); \
   # ensure both symbols are there
   if test $$(echo "$${PRINTFS}" | wc -l) = 2 ; then \
       echo "[FAILED] Expected printf and iprintf to be present, found $${PRINTFS}" ; \
       exit 1; \
   elif test $${PRINTF_ADDRS} != $(N_PRINTFS) ; then \ 
       echo "[FAILED] Expected $(N_PRINTFS) different printfs"; \
       exit 1; \
   fi
```

BTW, could we define a standard set of shell functions to be included in all makefiles. I would like to have a 'die' function:
```sh
die() {
    echo "${1}" >&2
    exit 1
}
# one-line "do stuff and exit with error if it fails"
test -e fileXX.txt || die "file not found"
```